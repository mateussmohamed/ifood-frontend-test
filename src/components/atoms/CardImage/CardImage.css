.card-image {
  border-radius: 5px;
}
