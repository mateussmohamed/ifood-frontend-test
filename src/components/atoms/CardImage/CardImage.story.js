import { storiesOf } from '@storybook/react';
import React from 'react';
import CardImage from './index';

const stories = storiesOf('CardImage', module);

const imageUrl =
  'http://img.wennermedia.com/920-width/rs-rihanna-anti-5f7d6f87-8c71-4389-b483-789925e27ff8.jpg';

stories
  .add('whit src prop and default width', () => <CardImage src={imageUrl} />)
  .add('whit src and width prop', () => (
    <div>
      <div style={{ float: 'left' }}>
        <h1>400x400</h1>
        <CardImage src={imageUrl} width="400" />
      </div>
      <div style={{ float: 'left' }}>
        <h1>200x200</h1>
        <CardImage src={imageUrl} width="200" />
      </div>
      <div style={{ float: 'left' }}>
        <h1>100x100</h1>
        <CardImage src={imageUrl} width="100" />
      </div>
    </div>
  ));
