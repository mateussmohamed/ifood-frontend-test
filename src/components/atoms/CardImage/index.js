import React from 'react';
import PropTypes from 'prop-types';
import './CardImage.css';

const CardImage = ({ src, title, width }) => (
  <img
    className="card-image"
    src={src}
    alt={title}
    title={title}
    width={width}
  />
);

CardImage.defaultProps = {
  title: '',
  width: 250,
};

CardImage.propTypes = {
  src: PropTypes.string.isRequired,
  title: PropTypes.string,
  width: PropTypes.number,
};

export default CardImage;
