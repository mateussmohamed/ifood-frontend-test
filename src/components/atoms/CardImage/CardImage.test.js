import React from 'react';
import renderer from 'react-test-renderer';
import CardImage from './index';

describe('CardImage', () => {
  test('should CardImage match to snapshot', () => {
    const imageURL = 'http://via.placeholder.com/400x400';
    const component = renderer.create(<CardImage src={imageURL} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('should width default prop equal 300', () => {
    const imageURL = 'http://via.placeholder.com/350x300';
    const component = renderer.create(<CardImage src={imageURL} />);
    expect(component.toTree().props.width).toBe(250);
  });
});
