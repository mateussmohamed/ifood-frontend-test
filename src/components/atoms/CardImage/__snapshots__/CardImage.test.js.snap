// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardImage should CardImage match to snapshot 1`] = `
<img
  alt=""
  className="card-image"
  src="http://via.placeholder.com/400x400"
  title=""
  width={300}
/>
`;
